File: 프로그래머스/연습문제/Lv1/옹알이 (2)/옹알이 (2).cpp
#include <string>
#include <vector>
using namespace std;

int solution(vector<string> babbling) {
    int cnt=babbling.size(), latest=-1;
    for(string ba : babbling){
        latest = -1;
        for(int i=0; i<ba.length(); i++){
            if(ba[i]=='a' && latest!=0){
                if(i>=ba.length()-2 || ba.substr(i,3)!="aya"){
                    cnt--; break;
                }
                else {
                    i += 2;
                    latest = 0;
                }
            }
            else if(ba[i]=='y' && latest!=1){
                if(i>=ba.length()-1 || ba.substr(i,2)!="ye"){
                    cnt--; break;
                }
                else{
                    i++;
                    latest = 1;
                }
            }
            else if(ba[i]=='w' && latest!=2){
                if(i>=ba.length()-2 || ba.substr(i,3)!="woo"){
                    cnt--; break;
                }
                else{
                    i += 2;
                    latest = 2;
                }
            }
            else if(ba[i]=='m' && latest!=3){
                if(i>=ba.length()-1 || ba.substr(i,2)!="ma"){
                    cnt--; break;
                }
                else{
                    i++;
                    latest = 3;
                }
            }
            else{
                cnt--; break;
            }
        }
    }
    
    return cnt;
}

File: 프로그래머스/연습문제/Lv1/옹알이 (2)/옹알이 (2).py
def solution(babbling):
    cnt=len(babbling); latest=-1
    for ba in babbling:
        latest=-1; i=0
        while i < len(ba):
            if ba[i]=='a' and latest!=0:
                if i>=len(ba)-2 or ba[i:i+3]!="aya":
                    cnt-=1; break
                else: i+=3; latest=0
            elif ba[i]=='y' and latest!=1:
                if i>=len(ba)-1 or ba[i:i+2]!="ye":
                    cnt-=1; break
                else: i+=2; latest=1
            elif ba[i]=='w' and latest!=2:
                if i>=len(ba)-2 or ba[i:i+3]!="woo":
                    cnt-=1; break
                else: i+=3; latest=2
            elif ba[i]=='m' and latest!=3:
                if i>=len(ba)-1 or ba[i:i+2]!="ma":
                    cnt-=1; break
                else: i+=2; latest=3
            else: cnt-=1; break

    return cnt
